garak/buffs/base.py

```diff
@@ -16,9 +16,11 @@
 import tqdm
 
 import garak.attempt
+from garak import _config
```

Comment:
It would be nice avoid the import not sure it would be a better experience for users or developers though.

Setting a default value for `config_root` to `None` could have allowed this. I went this direction to decouple the global config from the `Configurable` and hopefully provide a more clear interface for plugin consumers and creators to have control of the values.

Reply:
I like the quality of the current separation. Better decoupling was definitely needed.

The application of a standard param in so many constructors, including many "leaf" classes, and also the same line added to many of them, indicates there's potentially some benefit to setting a default behavior handled upstream. I think the expectation is for developers to inherit from base classes, where the notation tends to be clearer. Another expectation I think we have is that developing a new class isn't likely, generally, to involve customisation of how config is done.

On the other hand - it might that making this more minimal is tricky and fragile enough to outweigh the benefits. I don't know.

Reply:
> I think that's a great idea. It might also make sense for us to add _config to the base init.py file since it's used so widely?

👀👀👀

Reply:
I will defer on this item for the time being.